spec/spec_helper.rb

```diff
@@ -169,17 +169,23 @@ def thread_initialize
   def setup_filewatcher
     debug 'setup_filewatcher'
     debug filewatcher.inspect
-    filewatcher.watch do |filename, event|
+    filewatcher.watch do |changes|
       debug filewatcher.inspect
       @mutex.synchronize do
-        debug "watch callback: filename = #{filename}, event = #{event}"
+        log_changes(changes)
         increment_watched
-        @processed.push([filename, event])
+        @processed.push(changes)
         # debug 'pushed to processed'
       end
     end
   end
 
+  def log_changes(changes)
+    changes.each do |filename, event|
+      debug "watch callback: filename = #{filename}, event = #{event}"
+    end
+  end
```

Comment:
We can try to avoid this additional method with something like:

```ruby
debug "watch callback: changes = #{changes}" # maybe `changes.inspect`, but it will not be better as I know
```

What do you think?

Reply:
Sure, I can change that! I don't have much of an opinion either way, but your suggestion might make it easier to debug :)